Repl refactor

Qpirons opened the pull request:
Please code review this thank you.

willkill07 committed:
Model/CLI Refactor: Oh lawd she commin' :)

willkill07 committed:
Model/CLI Refactor: Oh lawd she commin' :)

lawilli4 committed:
Created 'Class.cpp' and its functions

Created the source file to match Professor Killian's 'Class.h' header file. Also added almost all functions introduced in the Class header file. Still missing 'rawName()'.

lawilli4 committed:
Added the rawName() function to Class.cpp

Qpirons committed:
Added the rest of the commands to the command factory, add edit delete etc.

Qpirons committed:
Added edit and delete commands to command.cpp

lawilli4 committed:
Merge pull request #129 from mucs420f19/update-class-files

Update class files

Qpirons committed:
Created Attribute.cpp and Relationship.cpp and added functions declarations to them

Qpirons committed:
Created Model.cpp and the relevent Function declarations, created function decs for Relationship Tostring

Qpirons committed:
Created Function Declarations for Attribute ToString

Qpirons committed:
Fixed Miss spelled variable name and function access call

Qpirons committed:
Included .cpp files and worked on errors missed by not including them earlier

Qpirons committed:
Fixed a few more errors found from including Command.cpp

Qpirons committed:
Included CommandFactory to work on compiler errors

Qpirons committed:
Fixed function call misnames

Qpirons committed:
Fixed compiler errors from virtual and override being in the wrong places

Qpirons committed:
Fixed some compiler errors from switch statement

Qpirons committed:
Fixed various compiler errors leaving only ones in command factory whcih are due to needing to write the commands

Qpirons committed:
Commented out skeleton code to allow for furthur testing

Qpirons committed:
Made minor changes to the Tostring functions

Qpirons committed:
changed i into being a reference and changed operator = function

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+#include "Attribute.h"
+
+Attribute::Attribute(const std::string & name, Visibility v)
+{
```
constructors should always use member initializer lists

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+	return ToString(mVisibility).c_str();
+}
+
+void Parameter::setType(const std::string & type)
```
Missing Parameter Constructor?

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+
+void Method::setParameters(const std::vector<Parameter>& params)
+{
+	for (auto & i : params)
```
This doesn't set -- it appends

just say mParameters = params;

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+
+void Method::clearParameters()
+{
+	int index = 0;
```
mParameters.clear() ?

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+
+const std::string & Method::params() const
+{
+	std::string params;
```
Not the right type of string you want. It also returns a const reference -- you are returning a temporary string. this will break the GUI

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,135 @@
+		params += "Type: " + i.type();
+		params += "Name: " + i.name() + ",";
+	}
+	return params.c_str();
```
Similar to above. Will not work

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+// Removes a field from the vector of fields after finding its index
+void Class::removeField(const std::string& name)
+{
+    unsigned int index = 0;
```
Need to use reference in for-each loop

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+void Class::removeMethod(const std::string& name)
+{
+    unsigned int index = 0;
+	for (auto i : mMethods)
```
use reference!

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+// Returns a method from the vector of methods given its name
+Method& Class::method(const std::string& name)
+{
+    for (auto i : mMethods)
```
Use reference...

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+// Returns a method from the vector of methods given its name
+const Method& Class::method(const std::string& name) const
+{
+    for (auto i : mMethods)
```
Use reference...

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+// Returns a field from the vector of fields gien its name
+Field& Class::field(const std::string& name)
+{
+    for (auto i : mFields)
```
Use reference...

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,143 @@
+// Returns a field from the vector of fields gien its name
+const Field& Class::field(const std::string& name) const
+{
+    for (auto i : mFields)
```
Use reference...

willkill07 commented on Class.h:
```diff
@@ -0,0 +1,37 @@
+
+private:
+  std::string mName;
+  std::vector<Field> mFields;
```
switch to std::list for both

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,69 @@
+}
+
+const std::vector<std::string> &Command::allCommands() {
+	//TODO
```
Still TODO?

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,69 @@
+					 "    - className: name for the new class\n"}
+
+      }};
+	//TODO
```
Still marked TODO? finish it

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,69 @@
+AddRelationshipCommand::AddRelationshipCommand(const std::string &parent,
+                                        const std::string &child,
+	const std::string &type)
+    : mParent(parent), mChild(child), mType(typeConvert(type)) {}
```
what is typeConvert?

willkill07 commented on Relationship.h:
```diff
@@ -0,0 +1,42 @@
+};
+
+
+RelationshipType typeConvert(const std::string &type);
```
would call this FromString

willkill07 commented on Relationship.cpp:
```diff
@@ -0,0 +1,86 @@
+
+const char * ToRawString(RelationshipType t)
+{
+	std::string temp;
```
This won't work. Apply what was done with the other ToString / ToRawString for attribute

willkill07 commented on Relationship.cpp:
```diff
@@ -0,0 +1,86 @@
+
+RelationshipType typeConvert(const std::string & type)
+{
+	std::string in = type;
```
mandate exact string matching

willkill07 replied:
And rename it to FromString

willkill07 replied:
Unresolved... still not mandating exact string matching

willkill07 commented on Model.h:
```diff
@@ -0,0 +1,29 @@
+  const Class* findClass (const std::string &name) const;
+
+private:
+  std::vector<Class> mClasses;
```
std::vector should be std::list so iterators / pointers are never invalidated

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+
+const Class * Model::findClass(const std::string & name) const
+{
+	for (auto  i : mClasses)
```
Reference...

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+
+Class * Model::findClass(const std::string & name)
+{
+	for (auto  i : mClasses)
```
Reference...

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+	Class c;
+	std::string pString;
+	std::string cString;
+	for (auto i : mRelationships)
```
Reference...

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+		pString = p.name();
+		c = i.child();
+		cString = c.name();
+		if (pString == one && cString == two)
```
should work with one, two and two, one

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+void Model::removeRelationship(const std::string & one, const std::string & two)
+{
+	int index = 0;
+	for (auto & i : mRelationships)
```
If you use iterators, this becomes way cleaner

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+	}
+}
+
+void Model::addRelationship(const std::string & parent, const std::string & child, RelationshipType type)
```
This just won't compile -- mRelationships doesn't hold pointers?

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,128 @@
+void Model::removeClass(const std::string & name)
+{
+	int index = 0;
+	for (auto & i : mClasses)
```
Iterators are your friend

willkill07 commented on Main.cpp:
```diff
@@ -8,24 +8,26 @@
-	UMLObjectsHolder* holder = new UMLObjectsHolder();
-
+	//UMLObjectsHolder* holder = new UMLObjectsHolder();
+	ConsoleView * holder = new ConsoleView(std::cout, std::cin);
```
why are you making a pointer?

willkill07 replied:
Also, why is called holder?

xcorniel95 committed:
adjusted files to have member initialization and better use of references. Converted from vector to list. Few issues that need addressed.

Qpirons committed:
Changed params to be adding to mStringrep and to return mStringrep

Qpirons committed:
Updated Class to Use member Initializer list, and changed rest of the functions to use iterators

Qpirons committed:
Added operator overload functions

Qpirons committed:
Added list command

Qpirons committed:
Added list command class

Qpirons committed:
Added call to List command

Qpirons committed:
Changed consoleview variable to not be a pointer and name to something that makes more sense

Qpirons committed:
Added list function,Fixed add class,Fixed removeclass, fixed addRelationship, made minor changes to other functions as well

Qpirons committed:
Added operator overload function and changed fromstring function

Qpirons committed:
Fixed setParameters to properly assign, Fixed clearParameters to properly clear

lawilli4 committed:
Added base code for help_for functionality

Added code to enable help_for functionality. A bug related to pointers currently exists when it is used.

lawilli4 committed:
Fixed bug relating to help_for function

Removed unneccesary set of brackets that were causing the error. Also added cout to the execute function to fix a logic error of not outputting the info to the user.

lawilli4 committed:
Added base code for the "help" command's allCommands function

It is currently incomplete and has a few bugs.

Qpirons committed:
Fixed errors keeping program from compiling

Qpirons committed:
edited allCommands and Helpcommand to properly print out commands from the vector

lawilli4 committed:
Added start-up display and updated text when an invalid command is entered

silviuburz committed:
MVC Refactor added functionality for fields

silviuburz committed:
MVC Refactor Implemented methods

silviuburz committed:
MVC Refactor added functionality for parameters

silviuburz committed:
MVC Refactor fix edit param declaration

silviuburz committed:
MVC Refactor implemented visibility and AddField in model

silviuburz committed:
MVC Refactor error checking for addfield

silviuburz committed:
MVC Refactor added fixes to addfield command

silviuburz committed:
MVC Refactor implemented saving/loading commands

silviuburz committed:
MVC Refactor implemented add_method

silviuburz committed:
Refactored FromString; implemented delete field + method

silviuburz committed:
MVC Refactor added in string getters to be used with tests

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,165 @@
+	return mName.c_str();
+}
+
+std::string Parameter::TestToString()
```
What is TestToString()?

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,165 @@
+
+const std::string & Method::params() const
+{
+	for (auto & i : mParameters)
```
This doesn't modify mStringRep... you can clear the string before appending to it (use +=)

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,165 @@
+
+std::string Method::TestToString()
+{
+	std::string out = "{" + name() + ", " + mReturnType + ", {";
```
Again, I'm not understanding what TestToString accomplishes... you have accessors. You can test each one. This is an arbitrary format that accomplishes nothing and only makes it more confusing to follow

willkill07 commented on Attribute.cpp:
```diff
@@ -0,0 +1,165 @@
+	Visibility result = Visibility::PRIVATE;
+	if (in.size() == 0) return result;
+	std::transform(std::cbegin(in), std::cend(in), std::begin(in), [](const unsigned char i) { return std::tolower(i); });
+	if (in == "private" || in[0] == '-') result = Visibility::PRIVATE;
```
I am confused by the usage of - + or #

Visibility should default to INVALID

and the only valid forms should be lowercase "private" "public" "protected"

willkill07 commented on Attribute.h:
```diff
@@ -0,0 +1,87 @@
+#include <cctype>
+
+
+enum class Visibility { PUBLIC = 0, PROTECTED = 1, PRIVATE = 2 };
```
You can introduce INVALID = -1

willkill07 commented on Attribute.h:
```diff
@@ -0,0 +1,87 @@
+  const std::string &type() const;
+  const char* rawType() const;
+
+  std::string TestToString();
```
Delete this

willkill07 commented on Attribute.h:
```diff
@@ -0,0 +1,87 @@
+  const char* rawType() const;
+  const char* rawName() const;
+
+  std::string TestToString();
```
Delete TestToString

willkill07 commented on Attribute.h:
```diff
@@ -0,0 +1,87 @@
+  const std::string& params() const;
+  const char* rawParams() const;
+
+  std::string TestToString();
```
Delete TestToString

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+void Class::removeField(int index)
+{
+	std::list<Field>::iterator it = mFields.begin();
+	std::advance(it, index);
```
Nice! 👍

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+	for (auto& i : mMethods)
+	{
+		if (i.name() == name)
+			result = &i;
```
you can just return `&i` here and change the return to nullptr below (result is unnecessary)

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+		if (i.name() == name)
+			result = &i;
+	}
+	return *result;
```
Wait... if this returns a reference, then you shouldn't be dereferencing nullptr. you have options here.

either: 1) change the return type to pointer
or: 2) keep as reference and throw an exception

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+}
+
+// Returns a method from the vector of methods given its name
+Method &Class::method(const std::string &name)
```
Is a precondition to this that the name is a name of a method?

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+// Returns a field from the vector of fields gien its name
+Field &Class::field(const std::string &name)
+{
+	Field* result = nullptr;
```
This is similar to the method problem above

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+	return *it;
+}
+
+std::string Class::TestFieldsToString()
```
Delete TestFieldsToString()

willkill07 commented on Class.cpp:
```diff
@@ -0,0 +1,207 @@
+	return out;
+}
+
+std::string Class::TestMethodsToString()
```
Delete TestMethodsToString()

willkill07 commented on Class.h:
```diff
@@ -0,0 +1,44 @@
+  Field& field (int index);
+  const Field& field (int index) const;
+
+  std::string TestFieldsToString();
```
Delete the two Test... methods

willkill07 commented on Class.h:
```diff
@@ -0,0 +1,44 @@
+  Method& method (int index);
+  const Method& method (int index) const;
+
+  Field& field (const std::string& name);
```
As mentioned previously in the implementation files, you have a choice to make with method(..) and field(..)

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+
+const std::vector<std::string> Command::allCommands() {
+	std::vector<std::string> myvector{ "help", "help_for", "list", "save", "load", "exit", "",
+	                                      "add_class", "add_field", "add_method", "add_parameter", "add_relationship", "",
```
Use tabs or spaces consistently. `clang-format` is amazing.

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+                     "  adds a new class to the UML diagram\n"
+                     "    parameters:\n"
+                     "    - className: name for the new class\n"}
+
```
more help commands/documentation should be here

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+	
+	auto out = HELP.find(name);
+
+	return (*out).second;
```
```suggestion
	if (out == HELP.end())
	{
		return "Invalid command";
	}
	return out->second;
```

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+
+void AddClassCommand::execute() const
+{
+	if (Command::modelInstance == nullptr)
```
This is not the place to initialize the modelInstance

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+
+void HelpCommand::execute() const
+{
+	std::vector<std::string> test = allCommands();
```
reference?

Qpirons replied:
If all commmands returns a refernce it wont print out anything for some reason so it was changed, if you can tell me why it wont print anything out if its a reference that would be greatly appreciated

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+
+void HelpForCommand::execute() const
+{
+	std::cout << helpFor(mName);
```
newline?

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+}
+
+AddMethodCommand::AddMethodCommand(const std::string& className, const std::string& methodName, const std::string& methodType, const std::string& methodVisibility)
+	: mclassName(className), mmethodName(methodName), mmethodType(methodType), mmethodVisibility(VisibilityFromString(methodVisibility)) {}
```
mClassName, mMethodName, mMethodVisibility

note the capitalization differences? mclass and mmethod are hard to read

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+}
+
+EditMethodCommand::EditMethodCommand(const std::string& whichAttr, const std::string& className, const std::string& methodName, const std::string& newValue)
+	: mclassName(className), mmethodName(methodName), mNewValue(newValue), mWhichAttr(whichAttr) {}
```
same as above...

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+	Command::modelInstance->deleteMethod(mclassName, mmethodName);
+}
+
+AddParameterCommand::AddParameterCommand(const std::string& className, const std::string& methodName, const std::string& ParameterName, const std::string& ParameterType)
```
I'm not going to continue to beat a dead horse here

willkill07 commented on Command.h:
```diff
@@ -0,0 +1,252 @@
+	virtual void execute() const override;
+
+private:
+	std::string mclassName, mmethodName, mParameterName, mParameterType;
```
naming conventions lowerCamelCase for variables

willkill07 commented on Command.h:
```diff
@@ -0,0 +1,252 @@
+
+class EditParameterCommand : public Command {
+public:
+	EditParameterCommand(const std::string& WhichAttr, const std::string& className, const std::string& methodName, const std::string& ParameterName, const std::string& NewValue);
```
parameter names should not be capitalized e.g. WhichAttr

willkill07 commented on Main.cpp:
```diff
@@ -8,24 +8,24 @@
-	UMLObjectsHolder* holder = new UMLObjectsHolder();
-
+	//UMLObjectsHolder* holder = new UMLObjectsHolder();
+	ConsoleView View = ConsoleView(std::cout, std::cin);
```
```suggestion
	ConsoleView view (std::cout, std::cin);
```

willkill07 commented on Main.cpp:
```diff
@@ -8,24 +8,24 @@
 	{
-		node_holder(holder);
-		RunGUI(holder);
+		View.runREPL();
```
```suggestion
		view.runREPL();
```

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,236 @@
+			i.rename(Newname);
+		}
+	}
+}
```
Error checking can happen here and return a bool (OR BETTER YET... an enum!)

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,236 @@
+void Model::addField(const std::string& className, const std::string& fieldName, const std::string& fieldType, Visibility fieldVisibility)
+{
+	Class* a = findClass(className);
+	if (a == nullptr) return;
```
You can report/return an error here

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,236 @@
+{
+	for (auto it = mRelationships.begin(); it != mRelationships.end(); ++it)
+	{
+		if ((*it).parent().name() == one && (*it).child().name() == two)
```
```suggestion
		if (it->parent().name() == one && it->child().name() == two)
```

willkill07 commented on Model.cpp:
```diff
@@ -0,0 +1,236 @@
+		{
+			mRelationships.erase(it);
+		}
+		if ((*it).parent().name() == two && (*it).child().name() == one)
```
Same fix here

willkill07 commented on Model.h:
```diff
@@ -0,0 +1,56 @@
+  Field* findField(const std::string& className, const std::string& fieldName);
+  Method* findMethod(const std::string& className, const std::string& methodName);
+
+  std::string TestClassToString(const std::string& className);
```
These Test... methods?

willkill07 commented on Model.h:
```diff
@@ -0,0 +1,56 @@
+
+  void removeRelationship(const std::string &one, const std::string &two);
+
+  void list();
```
list only makes sense for a console. not the GUI.

It does not belong in the model

willkill07 commented on Relationship.cpp:
```diff
@@ -0,0 +1,65 @@
+	mType = type;
+}
+
+std::string Relationship::TestToString() const
```
delete this

willkill07 commented on Relationship.h:
```diff
@@ -0,0 +1,48 @@
+const std::string &ToString(RelationshipType t);
+const char* ToRawString(RelationshipType t);
+
+bool operator== (const Relationship &n1, const Relationship &n2);
```
Move this up as part of the class.

```cpp
bool operator== (const Relationship &other) const;
```

Qpirons replied:
if i move that inside of the class i get a binary 'operator ==' has too many parameters error

willkill07 commented on Command.cpp:
```diff
@@ -0,0 +1,220 @@
+}
+
+const std::vector<std::string> Command::allCommands() {
+	std::vector<std::string> myvector{ "help", "help_for", "list", "save", "load", "exit", "",
```
@Qpirons this should return a const reference and the vector should be marked static inside of this function

Qpirons replied:
thank you

Qpirons committed:
Made changes based off of the code review

Qpirons committed:
Removed "test" functions as was requested

Qpirons committed:
Changed Variable names to be more readable and consistent

Qpirons committed:
MAde more changes based upon was was requested in the code review

Qpirons committed:
Removed test functions

Qpirons committed:
Fixed string issue that causes compile error

Qpirons committed:
Removed unnessecary string declaration

willkill07 commented on Model.cpp:
```diff
@@ -1,26 +1,26 @@
+
+void Model::addClass(const std::string & name)
+{
+	Class a(name);
```
You can add error checking to the model (by introducing return types)

willkill07 commented on Model.cpp:
```diff
@@ -1,26 +1,26 @@
+	{
+		if (i.name() == Oldname)
+		{
+			i.rename(Newname);
```
same here

willkill07 commented on Model.cpp:
```diff
@@ -75,162 +75,127 @@ void Model::editParameter(const std::string& className, const std::string& metho
+
+void Model::addRelationship(const std::string & parent, const std::string & child, RelationshipType type)
+{
+	auto p = find(mClasses.begin(), mClasses.end(), parent);
```
p may not exist. c may not exist. let the model dictate the type of error that should be presented to the user, but don't have the model print it or display it

willkill07 commented on Model.cpp:
```diff
@@ -75,162 +75,127 @@ void Model::editParameter(const std::string& className, const std::string& metho
+		{
+			mRelationships.erase(it);
+		}
+		if (it->parent().name() == two && it->child().name() == one)
```
shouldn't be both

willkill07 commented on Model.cpp:
```diff
@@ -75,162 +75,127 @@ void Model::editParameter(const std::string& className, const std::string& metho
 }
 
-Field* Model::findField(const std::string& className, const std::string& fieldName)
+void Model::list()
```
list() isn't part of Model -- it should only be a part of the view

willkill07 commented on Model.cpp:
```diff
@@ -75,162 +75,127 @@ void Model::editParameter(const std::string& className, const std::string& metho
 
-std::vector<const Relationship*> Model::findRelationship(const std::string& className) const
+void Model::removeRelationship(const std::string & one, const std::string & two)
 {
```
use findRelationship?

lawilli4 committed:
Worked on updating list functionality

Tried to recreate the old method of listing classes in order to better see our UML boxes in the REPL. Currently buggy, cannot get the code to print fields.

lawilli4 committed:
Added a few more commands to the "help_for" command

Qpirons committed:
Fixed issue preventing field and method from being created and modified list to output them

lawilli4 committed:
Updated the list command to properly print fields

Methods are not yet completed.

lawilli4 committed:
Created base code to print +/-/# symbols to represent visibility in UML boxes

Buggy, is not yet completed

lawilli4 committed:
Finshes UML box format in the REPL and fixed buggy +/-/# visibility display

lawilli4 committed:
Removed extra, unneccessary qualification

silviuburz committed:
fixed saving and loading

silviuburz committed:
fixed cmake

silviuburz committed:
fix build, still need to test functionality

silviuburz committed:
fix build

Qpirons committed:
Implemented loading into command.cpp

lawilli4 committed:
Worked on adding parameter functionality.

Code has errors and does not build. The function causing these issues is addParameter is Model.cpp

lawilli4 committed:
Fixed logic bug with addParameter so that it now builds

lawilli4 committed:
Added setName function for attributes, and setType functions for fields and methods

lawilli4 committed:
Added ability to edit fields and methods

Includes functionality to edit name, type, and visibility

lawilli4 committed:
Updated relationship display in the list command

lawilli4 committed:
Fixed a logic error that allowed the user to create two classes with the same name

lawilli4 committed:
Fixed a logic error that allowed the user to create a relationship between a class and itself

lawilli4 committed:
Fixed an error causing deleteRelationship to crash the program

silviuburz committed:
updated removeField

lawilli4 committed:
Attempted various bug fixes

lawilli4 committed:
Merge branch 'repl-refactor' of https://github.com/mucs420f19/Hercules into repl-refactor

Qpirons committed:
Changed mParameters to List and changed functions to work with that.

Qpirons committed:
Moved creating model into correct place, and fixed bug when causing crash when trying to delete non existent crash

Qpirons committed:
Fixed params function to no longer crash when called

Qpirons committed:
Re added function i removed by mistake and modified it, also modified deleteclass to delete relationships if you delete a class in a relationship

Qpirons committed:
Overloaded save command

lawilli4 committed:
Finalized the list command format

lawilli4 committed:
Added editParameter & deleteParameter

Still an error present relating to pointers

Qpirons committed:
Created const versions of some functions to fix some issues

Qpirons committed:
Fixed Extra Qualification error

willkill07 commented on ConsoleView.cpp:
```diff
@@ -10,6 +10,7 @@ ConsoleView::ConsoleView(std::ostream &os, std::istream &is) : os(os), is(is) {}
 void
 ConsoleView::runREPL() {
 
+  std::make_unique<StartCommand>();
```
I think you want to execute this, too?

lawilli4 committed:
Moved the logic of list out of the model and into the class and relationship files

bmekstro committed:
Added additional help_for functionality to the REPL.
